src/sentry/tagstore/base.py

```diff
@@ -94,3 +116,9 @@ def get_group_event_ids(self, project_id, group_id, tags):
         >>> get_group_event_ids(1, 2, {'key1': 'value1', 'key2': 'value2'})
         """
         raise NotImplementedError
+
+    def get_tag_value_qs(self, project_id, key, query=None):
```

Comment:
I'm not in love with returning a queryset, since it kind of defeats the purpose of the abstraction [0]... but I don't want to rewrite pagination as part of this change either.

[0] While our plan for Tags 2.0 is to continue to use Postgres/Django under the hood, in a perfect world a `tagstore` backend could be written in anything -- but for now we do require querysets.

Reply:
I agree for now. Good to note where it's being used, and I think we'll want to figure out if we need to maintain pagination or if we can provide some other light abstraction in the event of using an entirely different datastore hypothetically.